src/CleanSoftwareCron.php

```diff
@@ -191,21 +191,13 @@ protected static function deleteItems(
     ): int {
         global $DB;
 
-        $total = 0;
-
-        do {
-            $scope['LIMIT'] = min($max - $total, self::MAX_BATCH_SIZE);
-            $items = $DB->request($scope);
-            $count = count($items);
-            $total += $count;
+        $scope['LIMIT'] = min($max, self::MAX_BATCH_SIZE);
```

Comment:
`MAX_BATCH_SIZE` is made to limit query results, not to limit maximum items to delete.
For instance, you may want to delete 10000 items, and, in this case, it will do it in 5 batches of 2000 items.

Reply:
I will suggest another way to fix.